### PR TITLE
Glossary: argument to ascii

### PR DESCRIPTION
**Pull request by estelle**
Part of the great glossary review project, that remains undefined

Not sure why there are the extra commits, but they're not impacting the files changed, so i am going to pretend they aren't here

### CODE REVIEW

File: files/en-us/glossary/argument/index.md [hunk 1]
```diff
@@ -8,7 +8,22 @@ page-type: glossary-definition
...
+// This function takes a single value, so second argument passed is ignored
+function example(parameter) {
+  console.log(parameter); // Output = foo
+}
+```
+
 ## See also
 
-- [Difference between Parameter and Argument](<https://en.wikipedia.org/wiki/Parameter_(computer_programming)>) on Wikipedia
+- {{Glossary("parameter", "Difference between Parameter and Argument")}}
```
Comment (estelle) [inline files/en-us/glossary/argument/index.md:28]:
```suggestion
- {{Glossary("parameter", "Difference between parameter and argument")}}
```

File: files/en-us/glossary/argument/index.md [hunk 2]
```diff
@@ -8,7 +8,22 @@ page-type: glossary-definition
 
 An **argument** is a {{glossary("value")}} ({{Glossary("primitive")}} or {{Glossary("object")}}) passed as input to a {{Glossary("function")}}.
 
+The argument order within the function call should be the same order as the {{Glossary("parameter","parameters")}} in the function. When there are more arguments in a function call then parameters in the function definition, the extra arguments are ignored.
```
Comment (Josh-Cena) [inline files/en-us/glossary/argument/index.md:11]: The "When there are more arguments in a function call then parameters in the function definition, the extra arguments are ignored." part is JavaScript-specific, which I don't think is appropriate. We should, however, explicitly document the correlation between parameters and arguments.

File: files/en-us/glossary/array/index.md [hunk 3]
```diff
@@ -10,15 +10,18 @@ An _array_ is an ordered collection of data (either {{Glossary("primitive")}} or
 
 Each item in an array has a number attached to it, called a numeric index, that allows you to access it. In JavaScript, arrays start at index zero and can be manipulated with various {{Glossary("Method", "methods")}}.
 
-What an array in JavaScript looks like:
+Array in JavaScript look like this:
```
Comment (Josh-Cena) [inline files/en-us/glossary/array/index.md:13]:
```suggestion
Arrays in JavaScript look like this:
```

File: files/en-us/glossary/array/index.md [hunk 4]
```diff
@@ -10,15 +10,18 @@ An _array_ is an ordered collection of data (either {{Glossary("primitive")}} or
...
 
 ```js
+//Arrays in JavaScript can hold different types of data
 let myArray = [1, 2, 3, 4];
 let catNamesArray = ["Jacqueline", "Sophia", "Autumn"];
```
Comment (Josh-Cena) [inline files/en-us/glossary/array/index.md:18]:
```suggestion
// Arrays in JavaScript can hold different types of data
const myArray = [1, 2, 3, 4];
const catNamesArray = ["Jacqueline", "Sophia", "Autumn"];
```

File: files/en-us/glossary/argument/index.md (hunk 2 above)
Comment (estelle) [inline files/en-us/glossary/argument/index.md:11]:
```suggestion
The argument order within the function call should be the same as the {{Glossary("parameter","parameters")}} order in the function definition.
```

**Comment by hamishwillee** [comment]
> Not sure why there are the extra commits, but they're not impacting the files changed, so i am going to pretend they aren't here

FYI This sometimes happens when you make changes in main rather than a branch, at the time that the changes have not merged. If you decide you have to clean it you can do a fetch and rebase on upstream.

**Comment by estelle** [comment]
> > Not sure why there are the extra commits, but they're not impacting the files changed, so i am going to pretend they aren't here
> 
> FYI This sometimes happens when you make changes in main rather than a branch, at the time that the changes have not merged. If you decide you have to clean it you can do a fetch and rebase on upstream.

I spent at least 4 hours today fighting with git. i won in the end, but it was an epic battle.

File: files/en-us/glossary/ascii/index.md [hunk 5]
```diff
@@ -6,8 +6,12 @@ page-type: glossary-definition
 
 {{GlossarySidebar}}
 
-**ASCII** (_American Standard Code for Information Interchange_) is one of the most popular coding methods used by computers for converting letters, numbers, punctuation and control codes into digital form. Since 2007, {{Glossary("UTF-8")}} superseded it on the Web.
+**ASCII** (_American Standard Code for Information Interchange_) is a character encoding standard of 128 7-bit used by computers for converting letters, numbers, punctuation and control codes into digital form.
```
Comment (OnkarRuikar) [inline files/en-us/glossary/ascii/index.md:9]:
```suggestion
**ASCII** (_American Standard Code for Information Interchange_) is a character encoding standard of 128 7-bit used by computers for converting letters, numbers, punctuation, and control codes into digital form.
```

File: files/en-us/glossary/arpa/index.md [hunk 6]
```diff
@@ -6,9 +6,12 @@ page-type: glossary-definition
 
 {{GlossarySidebar}}
 
-**.arpa** (address and routing parameter area) is a {{glossary("TLD","top-level domain")}} used for Internet infrastructure purposes, especially reverse DNS lookup (i.e., find the {{glossary('domain name')}} for a given {{glossary("IP address")}}).
```
Comment (chrisdavidmills) [inline files/en-us/glossary/arpa/index.md:9]: I don't disagree with any of the text that is here, but the whole entry seems slightly off. This is specifically about ARPA the TLD, whereas ARPA was also the Advanced Research Projects Agency, the original name for [DARPA](https://en.wikipedia.org/wiki/DARPA). Maybe add a little bit of text to point this out, or create separate entries for disambiguation purposes, something like ARPA_TLD and APRA_Agency?

File: files/en-us/glossary/array/index.md [hunk 7]
```diff
@@ -10,15 +10,18 @@ An _array_ is an ordered collection of data (either {{Glossary("primitive")}} or
 
 Each item in an array has a number attached to it, called a numeric index, that allows you to access it. In JavaScript, arrays start at index zero and can be manipulated with various {{Glossary("Method", "methods")}}.
 
-What an array in JavaScript looks like:
+Arrays in JavaScript look like this:
```
Comment (chrisdavidmills) [inline files/en-us/glossary/array/index.md:13]: All your existing changes here look good. I'd also like you to do something about the sentences in the first paragraph — "Arrays are used to store multiple values in a single variable. This is compared to a variable that can store only one value." — which seem slightly off and possibly misleading.

How about "Arrays are used to store multiple values under a single variable name. A regular variable, on the other hand, can store only one value." ?

File: files/en-us/glossary/ascii/index.md (hunk 5 above)
Comment (chrisdavidmills) [inline files/en-us/glossary/ascii/index.md:9]:
```suggestion
**ASCII** (_American Standard Code for Information Interchange_) is a 128-code point, 7-bit character encoding standard used by computers for converting letters, numbers, punctuation, and control codes into digital form.
```

File: files/en-us/glossary/ascii/index.md [hunk 8]
```diff
@@ -6,8 +6,12 @@ page-type: glossary-definition
...
+
+ASCII has 128 code points. The first 33 points are non-printing control codes including the carriage return, line feed, tab, and several obsolete non-printable code stemming from its origin as telegraph codes. The other 95 are printable characters, including digits (0-9), lowercase (a-z) and uppercase (A-Z) letters, and punctuation symbols.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/ascii/index.md:11]:
```suggestion
The first 33 ASCII code points are non-printing control codes including the carriage return, line feed, tab, and several obsolete non-printable codes stemming from its origin of representing telegraph codes. The other 95 are printable characters, including digits (0-9), lowercase (a-z) and uppercase (A-Z) letters, and punctuation symbols.
```

File: files/en-us/glossary/ascii/index.md [hunk 9]
```diff
@@ -6,8 +6,12 @@ page-type: glossary-definition
...
+
+Many computer systems instead use {{glossary("Unicode")}}, which has millions of code points, but the first 128 of these are the same as the ASCII set. Since 2007, {{Glossary("UTF-8")}} superseded it on the Web.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/ascii/index.md:13]:
```suggestion
Many computer systems instead use {{glossary("Unicode")}}, which has millions of code points, but the first 128 of these are the same as the ASCII set. {{Glossary("UTF-8")}} superseded ASCII on the Web in 2007.
```

File: files/en-us/glossary/mime/index.md [hunk 10]
```diff
@@ -6,11 +6,10 @@ page-type: glossary-definition
 
 {{GlossarySidebar}}
 
-**MIME** "Multipurpose internet mail extensions" is a standard to describe documents in other forms beside ASCII text, e.g. audio, video and images.
+**MIME** "Multipurpose internet mail extensions" is a standard to describe documents in other forms besides ASCII text, e.g. audio, video, and images.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/mime/index.md:9]:
```suggestion
**MIME** (Multipurpose Internet Mail Extensions) is a standard to describe documents in other forms besides ASCII text, for example, audio, video, and images.
```

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Looking good @estelle. I've made a few suggestions, but overall this really improves these glossary entries.

File: files/en-us/glossary/argument/index.md [hunk 11]
```diff
@@ -8,7 +8,22 @@ page-type: glossary-definition
 
 An **argument** is a {{glossary("value")}} ({{Glossary("primitive")}} or {{Glossary("object")}}) passed as input to a {{Glossary("function")}}.
 
+The argument order within the function call should be the same as the {{Glossary("parameter","parameters")}} order in the function definition.
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/glossary/argument/index.md]: Added an example

File: files/en-us/glossary/argument/index.md [hunk 12]
```diff
@@ -8,7 +8,22 @@ page-type: glossary-definition
...
+// This function takes a single value, so second argument passed is ignored
+function example(parameter) {
+  console.log(parameter); // Output = foo
+}
+```
+
 ## See also
 
-- [Difference between Parameter and Argument](<https://en.wikipedia.org/wiki/Parameter_(computer_programming)>) on Wikipedia
+- {{Glossary("parameter", "Difference between parameter and argument")}}
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/glossary/argument/index.md:40]: i have to make this change in a different PR. My local git environment is somehow borked, and the issue has to do with this PR

**Comment by OnkarRuikar** [comment]
>  My local git environment is somehow borked, and the issue has to do with this PR

@estelle looks like you created this branch on top of [`fontcss2`](https://github.com/estelle/content/pull/325) branch and not `main`. This happened because you had `fontcss2` checked out and then you ran `git checkout -b glosssary4` so those commits from `fontcss2` came along in this branch as well. But when you merged that PR a new merge commit `4327e3d` was created in the `main` branch. So your branch is diverging due to these two commits.

 I always use following command to avoid such accident:
```bash
# create a new branch from 'main' branch. Note the explicit use of 'main'.
git checkout -b glossary4 main
```

If you are feeling brave you can do following to win the battle against `git` :)
```
git checkout main

# sync local main with `mdn/content`
# here 'upstream' is my remote `git@github.com:mdn/content.git (push)` : https://github.com/estelle/content/blob/glossary4/CONTRIBUTING.md#forking-and-cloning-the-repository
git pull upstream main && git push

git checkout glossary4

# note down the complete commit hash of commit prior to the first `font-variant-css2 data type` commit
git log

# Undo all the commits in this PR and get all the changes as modified files
git rest --soft <noted-hash-above>

# put all the changes in a stash
git stash save "glossary4"

# get latest commits from main
git rebase main

# get this PR changes back
git stash apply 0

# check the diffs to see if no extra change is there
# then make one final commit 
git add .
git commit -m "Glossary: argument to ascii"

# force push the changes to GitHub
git push --force

# on this PR page you'll be left with only one commit containing all the changes done so far
```

File: files/en-us/glossary/arpa/index.md [hunk 13]
```diff
@@ -6,9 +6,12 @@ page-type: glossary-definition
...
+**.arpa** (address and routing parameter area) is a {{glossary("TLD","top-level domain")}} in the Domain Name System (DNS) used for Internet infrastructure purposes, especially reverse DNS lookup (i.e., find the {{glossary('domain name')}} for a given {{glossary("IP address")}}).
```
Comment (estelle) [inline files/en-us/glossary/arpa/index.md:9]:
```suggestion
**.arpa** (address and routing parameter area) is a {{glossary("TLD","top-level domain")}} in the Domain Name System (DNS) used for Internet infrastructure purposes, especially reverse DNS lookup (i.e., find the {{glossary('domain name')}} for a given {{glossary("IP address")}}).

The name comes from **ARPA**, currently known as [DARPA](https://en.wikipedia.org/wiki/DARPA), the Defense Advanced Research Projects Agency. DARPA's is credited with developing precursors to the Internet ({{glossary("ARPANET")}}), GPS, artificial intelligence, and virtual reality. 

{{glossary("TLD")}} was designed at DARPA during a time when it was called ARPA, hence the `.arpa` domain name.
```

**Comment by estelle** [comment]
THanks Onkar for https://github.com/mdn/content/pull/28718#issuecomment-1697769916. My local git is working again, and after all the headache yesterday i am not brave enough to try fixing it, but I have bookmarked your comment for next time.

File: files/en-us/glossary/argument/index.md [hunk 14]
```diff
@@ -8,7 +8,34 @@ page-type: glossary-definition
...
```
Comment (estelle) [inline files/en-us/glossary/argument/index.md:40]:
```suggestion
- [Difference between parameter and argument](/en-US/docs/Glossary/Parameter#parameters_versus_arguments)
```

File: files/en-us/glossary/argument/index.md [hunk 15]
```diff
@@ -8,7 +8,34 @@ page-type: glossary-definition
 
 An **argument** is a {{glossary("value")}} ({{Glossary("primitive")}} or {{Glossary("object")}}) passed as input to a {{Glossary("function")}}.
 
+For example:
+
+```js
+const argument1 = "Web";
+const argument2 = "Development;
```
Comment (chrisdavidmills) [inline files/en-us/glossary/argument/index.md:15]: Missing a closing quote here.

File: files/en-us/glossary/argument/index.md [hunk 16]
```diff
@@ -8,7 +8,34 @@ page-type: glossary-definition
...
+example(argument1, argument2); // passing two arguments
+
+// This function takes two values
+function example(parameter1, parameter2) {
+  console.log(parameter1); // Output = "Web"
+  console.log(parameter2); // Output = "Development
```
Comment (chrisdavidmills) [inline files/en-us/glossary/argument/index.md:21]: And here.

File: files/en-us/glossary/argument/index.md [hunk 17]
```diff
@@ -8,7 +8,34 @@ page-type: glossary-definition
...
+```
+
+The argument order within the function call should be the same as the {{Glossary("parameter","parameters")}} order in the function definition.
+
+```js
+const argument1 = "foo";
+const argument2 = [1, 2, 3];
+example(argument1, argument2); // passing two arguments
+
+// This function takes a single value, so second argument passed is ignored
```
Comment (chrisdavidmills) [inline files/en-us/glossary/argument/index.md:32]:
```suggestion
// This function takes a single value, so the second argument passed is ignored
```

File: files/en-us/glossary/arpa/index.md [hunk 18]
```diff
@@ -6,9 +6,16 @@ page-type: glossary-definition
...
+
+The name comes from **ARPA**, currently known as [DARPA](https://en.wikipedia.org/wiki/DARPA), the Defense Advanced Research Projects Agency. DARPA's is credited with developing precursors to the Internet ({{glossary("ARPANET")}}), GPS, artificial intelligence, and virtual reality.
```
Comment (chrisdavidmills) [inline files/en-us/glossary/arpa/index.md:11]:
```suggestion
The name comes from **ARPA**, currently known as [DARPA](https://en.wikipedia.org/wiki/DARPA), the Defense Advanced Research Projects Agency. DARPA is credited with developing precursors to the Internet ({{glossary("ARPANET")}}), GPS, artificial intelligence, and virtual reality.
```

**Review by chrisdavidmills — APPROVED** [review]
Nice work @estelle! I've found a couple more nits, but approving so as not to hold you up.

File: files/en-us/glossary/argument/index.md (hunk 15 above)
Comment (estelle) [inline files/en-us/glossary/argument/index.md:15]:
```suggestion
const argument2 = "Development";
```

File: files/en-us/glossary/argument/index.md [hunk 19]
```diff
@@ -8,7 +8,34 @@ page-type: glossary-definition
...
+
+```js
+const argument1 = "Web";
+const argument2 = "Development";
+example(argument1, argument2); // passing two arguments
+
+// This function takes two values
+function example(parameter1, parameter2) {
+  console.log(parameter1); // Output = "Web"
+  console.log(parameter2); // Output = "Development
```
Comment (estelle) [inline files/en-us/glossary/argument/index.md:21]:
```suggestion
  console.log(parameter2); // Output = "Development"
```

**Pull request merged**